File: sentimental-cash/cash.py
from cs50 import get_float


def main():
    while True:
        try:
            total = get_float("Change owed: ")
            if total > 0:
                break
        except ValueError:
            print("That's not a float.")

    # Number of coins
    coins = get_coins(total)
    print(f"{coins}")


def get_coins(total):
    # Number of quarters
    quarters = total // 0.25
    total = round(total - quarters * 0.25, 2)
    # Number of dimes
    dimes = total // 0.10
    total = round(total - dimes * 0.10, 2)
    # Number of nickels
    nickels = total // 0.05
    total = round(total - nickels * 0.05, 2)
    # Number of pennies
    pennies = total // 0.01
    coins = int(quarters + dimes + nickels + pennies)
    return coins


if __name__ == "__main__":
    main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    // How many cents do you own?
    int cents;
    do
    {
        cents = get_int("How many cents do you own? ");
    }
    while (cents < 0);
    return cents;
}

int calculate_quarters(int cents)
{
    // Count the number of quarters
    int quarters;
    quarters = cents / 25;
    return quarters;
}

int calculate_dimes(int cents)
{
    // Count the number of dimes
    int dimes;
    dimes = cents / 10;
    return dimes;
}

int calculate_nickels(int cents)
{
    // Count the number of nickels
    int nickels;
    nickels = cents / 5;
    return nickels;
}

int calculate_pennies(int cents)
{
    // Count the number of pennies
    int pennies;
    pennies = cents / 1;
    return pennies;
}
